### PR TITLE
Fixes #21217 - Handle empty string proxy parameters

### PR DESCRIPTION
**Pull request by ekohl**
Also fixes the migration tests and moves a lost capsule migration. Wasn't fully sure if I should include those in this PR.

### CODE REVIEW

File: config/katello.migrations/171005163313-empty-proxy-strings.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+if answers['katello']
+  ['url', 'port', 'username', 'password'].each do |name|
+    key = "proxy_#{name}"
+    if answers['katello'][key] == ''
```
Comment (ehelms) [inline config/katello.migrations/171005163313-empty-proxy-strings.rb]: Ahh, and so after this point '' will never be a valid string again? Unless I as a user go edit the answer file directly.

Comment (ekohl) [reply config/katello.migrations/171005163313-empty-proxy-strings.rb]: Currently the katello module refuses to use them so this clears them which used to be the same. If we do start accepting empty strings again we can remove this migration.

**Comment by ehelms** [comment]
Looks good -- mind squashing down to a single commit ?

**Comment by ehelms** [comment]
Looking good, can squash down to just the proper commit?

**Comment by ekohl** [comment]
Updated

**Pull request merged**